twenty-first/src/math/tip5.rs

```diff
@@ -673,10 +675,39 @@ impl Sponge for Tip5 {
     }
 }
 
+impl Hasher for Tip5 {
+    fn finish(&self) -> u64 {
+        self.state[0].value()
+    }
+
+    fn write(&mut self, bytes: &[u8]) {
+        let bfield_elements = bytes.chunks(BFieldElement::BYTES).map(|chunk| {
+            let mut buffer = [0u8; BFieldElement::BYTES];
+            buffer[..chunk.len()].copy_from_slice(chunk);
+            BFieldElement::new(u64::from_le_bytes(buffer))
+        });
+
+        let binding: Vec<BFieldElement> = bfield_elements.collect();
+        let bfield_absorb_chunks = binding.chunks(Tip5::RATE);
```

Comment:
These bindings are not really necessary. Their awkward names suggest that they don't really serve any purpose other than trying to make the types work.

How about starting the second loop in the following way instead?

```rust
for chunk in bfield_elements.chunks(Tip5::RATE).into_iter() {
	// …
}
```

For some reason, the explicit `.into_iter()` seems to be necessary here, even though it usually is not necessary for `for` loops. I guess `itertools::Itertools::IntoChunks` is a little odd?